pentagon/cli.py

```diff
@@ -65,18 +74,111 @@ def start_project(name, **kwargs):
 
 
 @click.command()
-@click.option('--name', help="Component name", required=True)
-@click.option('--environment', help="Environment")
-@click.option('--cluster', help="Cluster name")
-@click.option('--namespace', help="Namespace")
-@click.option('--version', help="Version")
-def install_component(name, **kwargs):
+@click.pass_context
+@click.argument('component_path')
+@click.option('--data', '-D', multiple=True, help='Individual Key=Value pairs used by the component')
+@click.option('--file', '-f', help='File to read Key=Value pair from (yaml or json are supported)')
+@click.option('--out', '-o', default='./', help="Path to output module result, if any")
+@click.option('--log-level', default="INFO", help="Log Level DEBUG,INFO,WARN,ERROR")
+@click.argument('additional-args', nargs=-1, default=None)
+def add(ctx, component_path, additional_args, **kwargs):
+    _run('add', component_path, additional_args, kwargs)
+
+
+@click.command()
+@click.pass_context
+@click.argument('component_path')
+@click.option('--data', '-D', multiple=True, help='Individual Key=Value pairs used by the component')
+@click.option('--file', '-f', help='File to read Key=Value pair from (yaml or json are supported)')
+@click.option('--out', '-o', default='./', help="Path to output module result, if any")
+@click.option('--log-level', default="INFO", help="Log Level DEBUG,INFO,WARN,ERROR")
+@click.argument('additional-args', nargs=-1, default=None)
+def get(ctx, component_path, additional_args, **kwargs):
+    _run('get', component_path, additional_args, kwargs)
```

Comment:
I really like that you're decoupling the install path from the component itself :+1: That should make this more flexible if we need to rejigger the layout!

Reply:
Thats what I was going for. I know it will cause some pain as we figure out how we _actually_ use them, but for now, they are totally flexible.